bundles/org.openhab.binding.somfymylink/src/main/java/org/openhab/binding/somfymylink/internal/handler/SomfyMyLinkBridgeHandler.java

```diff
@@ -0,0 +1,434 @@
+        if (config == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "mylink configuration missing");
+            return false;
+        }
+
+        if (StringUtils.isEmpty(config.ipAddress) || StringUtils.isEmpty(config.systemId)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "mylink address or system id not specified");
+            return false;
+        }
+
+        return true;
+    }
+
+    private void cancelHeartbeat() {
+        logger.debug("Stopping heartbeat");
+        ScheduledFuture<?> heartbeat = this.heartbeat;
+
+        if (heartbeat != null) {
+            logger.debug("Cancelling heartbeat job");
+            heartbeat.cancel(true);
+            this.heartbeat = null;
+        } else {
+            logger.debug("Heartbeat was not active");
+        }
+    }
+
+    private void sendHeartbeat() {
+        try {
+            logger.debug("Sending heartbeat");
+
+            SomfyMyLinkConfiguration config = this.config;
+            if (config == null) {
+                throw new SomfyMyLinkException("Config not setup correctly");
+            }
+
+            SomfyMyLinkCommandShadePing command = new SomfyMyLinkCommandShadePing(config.systemId);
+            sendCommandWithResponse(command, SomfyMyLinkPingResponse.class).get();
+            updateStatus(ThingStatus.ONLINE);
+
+        } catch (SomfyMyLinkException | InterruptedException | ExecutionException e) {
+            logger.warn("Problem with mylink during heartbeat: {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    public SomfyMyLinkShade[] getShadeList() throws SomfyMyLinkException {
+        SomfyMyLinkConfiguration config = this.config;
+        if (config == null || StringUtils.isEmpty(config.systemId)) {
+            throw new SomfyMyLinkException("Config not setup correctly. System Id is not set.");
+        }
+
+        SomfyMyLinkCommandShadeList command = new SomfyMyLinkCommandShadeList(config.systemId);
+
+        try {
+            SomfyMyLinkShadesResponse response = sendCommandWithResponse(command, SomfyMyLinkShadesResponse.class)
+                    .get();
+
+            if (response != null) {
+                return response.getResult();
+            } else {
+                return EMPTY_SHADE_LIST;
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            throw new SomfyMyLinkException("Problem while getting shade list.", e);
+        }
+    }
+
+    public SomfyMyLinkScene[] getSceneList() throws SomfyMyLinkException {
+        SomfyMyLinkConfiguration config = this.config;
+        if (config == null || StringUtils.isEmpty(config.systemId)) {
+            throw new SomfyMyLinkException("Config not setup correctly. System Id is not set.");
+        }
```

Comment:
```suggestion
        if (StringUtils.isEmpty(config.systemId)) {
            throw new SomfyMyLinkException("Config not setup correctly. System Id is not set.");
        }
```

Please make the similar changes in the other methods

Reply:
done